### PR TITLE
Added support for private vlan config in VMware distributed port group

### PR DESCRIPTION
**Pull request by axarriola**
##### SUMMARY
Added support for creating a distributed port group with a private VLAN.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
/lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup.py

##### ADDITIONAL INFORMATION
Tested on vSphere 6.7

### CODE REVIEW

**Comment by mariolenz** [comment]
Looks like a duplicate of #65256... admittedly, this PR looks somewhat abandoned.

Anyway, I guess the ansible team will want to have a [changelog fragment](https://docs.ansible.com/ansible/2.9/community/development_process.html#changelogs) and a test case.

**Comment by axarriola** [comment]
Thanks for your prompt review. I added the changelog fragment and a few test cases, which I'm troubleshooting. For some reason no pvlan is present in the dvswitch, even though the task to create them returned changed and a success msg.

**Comment by mariolenz** [comment]
> For some reason no pvlan is present in the dvswitch, even though the task to create them returned changed and a success msg.

Does it work in your environment but shippable fails? In that case, it might be a problem with shippable. It uses only a simulator (govcsim, I think) which sometimes leads to false negatives. That's why the ansible team works on a new CI pipeline where the tests run against a real vCenter. This new CI pipeline [looks OK](https://object-storage-ca-ymq-1.vexxhost.net/v1/a0b4156a37f9453eb4ec7db5422272df/ansible_05/67205/e99fd093c753829e5adea4a533d0c9ac3cd063e4/third-party-check/ansible-test-cloud-integration-vcenter_only-python36/2985e5a/controller/ara-report/reports/5ce94ca6-b814-4f5e-a825-624ab76a6527.html) to me. If this is really a shippable / govcsim problem, I think the best way would be to run the tests in a `when: vcsim is not defined` block.

@goneri Can you have a look at this? I'm quite sure this is a problem with the simulator, but I'd like to have a second opinion.

**Comment by axarriola** [comment]
Because the creation of a pvlan portgroup with vmware_dvs_portgroup depends on a private vlan already configured in the dvs (manually or by using vmware_dvswitch_pvlans), I basically copied the [integration test for vmware_dvswitch_pvlans](https://github.com/ansible/ansible/blob/devel/test/integration/targets/vmware_dvswitch_pvlans/tasks/main.yml) in the vmware_dvs_portgroup test.
I also added a vmware_dvs_portgroup_facts task to check the vlans configured in the dvs after the vmware_dvswitch_pvlans task. The vmware_dvswitch_pvlans task executes succesfully, but the supposedly newly configured private vlans do not appear in the vmware_dvs_portgroup_facts output. I guess the tests for vmware_dvswitch_pvlans are a false positive?

File: changelogs/fragments/67205-pvlan-config-vmware-dvs-portgroup.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+- vmware_dvs_portgroup - Added support for dpg with private VLAN.
```
Comment (goneri) [inline changelogs/fragments/67205-pvlan-config-vmware-dvs-portgroup.yml]: `dpg` is a bit confusing. Could you write `distributed port group` instead?

File: test/integration/targets/vmware_dvs_portgroup/tasks/main.yml [hunk 2]
```diff
@@ -385,3 +385,136 @@
   assert:
     that:
         - dvs_pg_result_0017.changed
+
```
Comment (goneri) [inline test/integration/targets/vmware_dvs_portgroup/tasks/main.yml:388]: Agreed with @mariolenz, you can just use a `when: vcsim is not defined`. You can either use a `block:` or, move all the tasks in a new playbook and do a:

```
- import_tasks: private_vlan.yaml
  when: vcsim is not defined
```

**Review by goneri — APPROVED** [review]
Looks good to me, we just need to resolve the govcsim issue. Thanks for the patch @axarriola!

File: test/integration/targets/vmware_dvs_portgroup/tasks/main.yml (hunk 2 above)
Comment (axarriola) [reply test/integration/targets/vmware_dvs_portgroup/tasks/main.yml:388]: Thanks for the tip! I used a block and also removed the extra vmware_dvs_portgroup_facts debug.

File: changelogs/fragments/67205-pvlan-config-vmware-dvs-portgroup.yml (hunk 1 above)
Comment (axarriola) [reply changelogs/fragments/67205-pvlan-config-vmware-dvs-portgroup.yml]: Of course, it's changed now.

**Comment by axarriola** [comment]
Thanks for your prompt responses and help! @goneri @mariolenz 

I made the final change requests and pushed them.

**Review by goneri — APPROVED** [review]
LGTM and the tests pass

**Comment by goneri** [comment]
@Akasurde can we merge this PR?

**Comment by goneri** [comment]
Hi!

Thank you for your PR. Since March 2, 2020, the Ansible team does not accept any new changes on the VMware modules on the core `ansible/ansible` repository.
The development continue but in the new VMware collection repository: https://github.com/ansible-collections/vmware
Could you please reopen the PR there?

Links:
- https://groups.google.com/forum/#!msg/ansible-devel/6UAfogEDtG8/wMTgxBGWCgAJ
- https://github.com/ansible-collections/overview/blob/master/README.rst

**Pull request closed without merging**